### PR TITLE
Remove rows with duplicate query cols and cleanup handling of TFM variants

### PR DESCRIPTION
**Pull request by siddharth-krishna**
This PR makes the following changes:
- It implements the [suggestion](https://github.com/etsap-TIMES/xl2times/pull/166#discussion_r1466952849) of using `Config.known_columns` of the table's tag instead of always using that of TFM_INS.
- It adds `TFM_INS-TXT` to `veda-tags.json` and modifies `Config` to use the base tag's known columns if none is specified. (These two changes correspond to the first 3 commits, and introduce no regression.)
- It removes rows with duplicate query columns (for GAMS parameters), which is [needed]( https://github.com/etsap-TIMES/xl2times/pull/166#issuecomment-1918683939 ) because #166 changed the behaviour of TFM_UPD to add new rows instead of updating rows inplace.
- However, this leads to a regression of correct rows, because for example in Demo 9 the ground truth DD repo has COM_PROJ defined in multiple files, with files having duplicate rows (when considering only query columns). So `dd_to_csv.py` is producing a CSV file for COM_PROJ that has duplicate rows, whereas we now only keep the last row in our output.
- I tried to resolve the above regression by modifying the ground truth comparison code to first remove duplicates in the ground truth before comparing. I then manually checked that the benchmarks that continue to regress only do so because additional rows or ground truth rows reduced, and not because missing rows increased.

However, Ireland has new missing rows:
image

I looked into the new missing rows in ACT_EFF in Ireland:
image
image

It looks like the problem is that the order in which `dd_to_csv.py` reads DD files puts the `1` row last, while we produce  the `1.075` row last. Perhaps we need to add the DD file order to `benchmarks.yml` so that we read DD files in the same order that GAMS will?

### CODE REVIEW

**Comment by olejandro** [comment]
@siddharth-krishna for the moment I believe we only should be removing duplicate rows for the case described [here](https://github.com/etsap-TIMES/xl2times/pull/166#issuecomment-1919600500). (Sorry haven't looked at the changes yet, so guessing that removal is not limited to the table that generates the rows)

**Comment by siddharth-krishna** [comment]
Do you mean we should only remove duplicate rows that are generated by the same TFM_UP table?

Why not remove the duplicates in all final outputs? I thought GAMS only considers the last one anyway.

**Comment by olejandro** [comment]
@siddharth-krishna, in addition to the above, how about we drop all the duplicates but only when printing to GAMS? This won't "overwrite" correctly yet, but the GAMS tests will be working.

**Comment by siddharth-krishna** [comment]
> remove duplicate rows that are generated by the same TFM_UP table

Haha we should be more precise in our sentences. Even this can mean two things: now that TFM_UPD adds new rows to the table instead of updating in place, it could mean (a) if a TFM_UPD table generates 4 rows, out of which 2 have the same query columns, then only add 1 of the dupicated rows; or (b) the table generates 4 rows, all of which have corresponding duplicate rows in the original FI_T table, so delete the original rows and add the 4 new rows in, but if a second table generates 2 rows that overlap with the previous 4, do not delete the original ones but add 2 new rows.

I tried to implement (a), but on Demo 4 that still leaves us with 3 additional rows (the original ACT_COST = 1111 rows).

Implementing (b) without in-place updates is tricky! We'll have to keep track of which rows came from where..

Let's talk more on our call. For now, I've moved the de-duplicating code to the write DD function, and kept the behaviour of TFM_UPD to always add new rows.

**Comment by Antti-L** [comment]
> I tried to implement (a), but on Demo 4 that still leaves us with 3 additional rows (the original ACT_COST = 1111 rows).

I am seeing them **hard-coded** in your code, `def generate_dummy_processes`. :grin:   Apparently VEDA2 does not write out such "internal" ACT_COST default values for the dummy imports (VEDA-FE actually did).

**Comment by siddharth-krishna** [comment]
@Antti-L ah yes, thanks, so that's where it comes from. I guess we should remove them before writing output, I've made an issue.

Based on my discussion with @olejandro the plan is to essentially merge this PR as it is. Rows with duplicate indices/query-cols are now removed just before writing DD files. This means the GDXDIFF test for Demos 1-4 are at 100% (OK). The other test shows 4 additional rows, but that can be solved by removing dummy rows, and by adding support for `sourcescen` -- both tasks for future PRs. :)

File: xl2times/config/veda-tags.json [hunk 1]
```diff
@@ -1664,6 +1664,18 @@
...
     "mod_type": "tsl"
   },
+  {
+    "tag_name": "tfm_ins-txt",
+    "tag_allowed_in": [
+      "BY_Trans",
+      "SR_Trans",
+      "SysSettings",
+      "RegScen",
+      "TradeScen"
```
Comment (olejandro) [inline xl2times/config/veda-tags.json:1674]:
```suggestion
      "SysSettings"
```

Comment (olejandro) [reply xl2times/config/veda-tags.json:1674]: My understanding is that this is only available in "SysSettings" under Veda. @Antti-L do you have a similar impression?

E.g.: https://forum.kanors-emr.org/showthread.php?tid=1245&highlight=%22tfm_ins-txt%22

**Review by olejandro — APPROVED** [review]

**Pull request merged**